sdk-cpp/tests/test_helpers.h

```diff
@@ -8,21 +8,146 @@
+    }
+    void resume()
+    {
+        std::error_code ec = _downloadImpl->resume();
+        msdod::throw_if_fail(ec);
+    }
+    void finalize()
+    {
+        std::error_code ec = _downloadImpl->finalize();
+        msdod::throw_if_fail(ec);
+    }
+    void abort()
+    {
+        std::error_code ec = _downloadImpl->abort();
+        msdod::throw_if_fail(ec);
+    }
+    msdo::download_status get_status() const
+    {
+        msdo::download_status status;
+        std::error_code ec = _downloadImpl->get_status(status);
+        msdod::throw_if_fail(ec);
+        return status;
+    }
+
+    void start_and_wait_until_completion(std::chrono::seconds timeoutSecs = std::chrono::hours(24))
+    {
+        std::error_code ec = _downloadImpl->start_and_wait_until_completion(timeoutSecs);
+        msdod::throw_if_fail(ec);
+    }
+    void start_and_wait_until_completion(const std::atomic_bool& isCancelled, std::chrono::seconds timeoutSecs = std::chrono::hours(24))
+    {
+        std::error_code ec = _downloadImpl->start_and_wait_until_completion(isCancelled, timeoutSecs);
+        msdod::throw_if_fail(ec);
+    }
+
+    static void download_url_to_path(const std::string& uri, const std::string& downloadFilePath,
+        std::chrono::seconds timeoutSecs = std::chrono::hours(24))
+    {
+        std::error_code ec = msdo::download::download_url_to_path(uri, downloadFilePath, timeoutSecs);
+        msdod::throw_if_fail(ec);
+    }
+    static void download_url_to_path(const std::string& uri, const std::string& downloadFilePath, const std::atomic_bool& isCancelled,
+        std::chrono::seconds timeoutSecs = std::chrono::hours(24))
+    {
+        std::error_code ec = msdo::download::download_url_to_path(uri, downloadFilePath, isCancelled, timeoutSecs);
+        msdod::throw_if_fail(ec);
+    }
+
+    /*
+    For devices running windows before 20H1, dosvc exposed a now-deprecated com interface for setting certain download properties.
+    After 20H1, these properties were added to newer com interface, which this SDK is using.
+    Attempting to set a download property on a version of windows earlier than 20H1 will not set the property and throw an exception with error code msdo::errc::do_e_unknown_property_id
+    */
+    void set_property(msdo::download_property key, const msdo::download_property_value& value)
+    {
+        std::error_code ec = _downloadImpl->set_property(key, value);
+        msdod::throw_if_fail(ec);
+    }
+    msdo::download_property_value get_property(msdo::download_property key)
+    {
+        msdo::download_property_value propValue;
+        std::error_code ec = _downloadImpl->get_property(key, propValue);
+        msdod::throw_if_fail(ec);
+        return propValue;
+    }
+
+private:
+    std::unique_ptr<msdo::download> _downloadImpl;
+};
+
+}
+}
+}
+
 class TestHelpers
 {
 public:
-    static void CleanTestDir();
-    static void CleanupWorkingDir();
+    static void CleanTestDir()
```

Comment:
you're using C++17, right?  if so, boost isn't needed.

`std::filesystem::exists`, `std::filesystem::remove`

Reply:
Still on C++14.